test/testcommon/src/main/java/io/pravega/test/common/CollectingExecutor.java

```diff
@@ -159,7 +159,7 @@ public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command, long initialD
     }
 
     public List<Runnable> getScheduledTasks() { 
-        return Collections.unmodifiableList(tasks);
+        return new ArrayList<>(tasks);
```

Comment:
is it okay to convert Collections.unmodifiableList -> ArrayList ?

Reply:
Unmodifiable list is a wrapper which is a view of the underlying list but through which modifications cannot be made. ArrayList is just making a copy of the list. So in the former case we know the caller isn't modifying the list but this class still could causing problems for the caller. In the case of a copy it allows the two lists to be independent, which since we know the caller of getScheduledTasks is not modifying the list an expecting it to be updated here, is equivalent and more conservative.

Reply:
ok.